parquet-hadoop/src/main/java/org/apache/parquet/crypto/AesDecryptor.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.crypto;
+
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import static org.apache.parquet.crypto.AesEncryptor.NONCE_LENGTH;
+import static org.apache.parquet.crypto.AesEncryptor.GCM_TAG_LENGTH;
+import static org.apache.parquet.crypto.AesEncryptor.CTR_IV_LENGTH;
+import static org.apache.parquet.crypto.AesEncryptor.CHUNK_LENGTH; 
+import static org.apache.parquet.crypto.AesEncryptor.INT_LENGTH;
+
+import org.apache.parquet.crypto.AesEncryptor.Mode;
+import org.apache.parquet.format.BlockCipher;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+
+
+public class AesDecryptor implements BlockCipher.Decryptor{
+
+  private final Mode aesMode;
+  private final SecretKey aesKey;
+  private final int tagLength;
+  private final Cipher aesCipher;
+  private final byte[] ctrIV;
+  private final byte[] nonce;
+
+  /**
+   * 
+   * @param mode GCM or CTR
+   * @param keyBytes encryption key
+   * @throws IllegalArgumentException
+   * @throws IOException
+   */
+  public AesDecryptor(Mode mode, byte[] keyBytes) throws IllegalArgumentException, IOException {
+    if (null == keyBytes) {
+      throw new IllegalArgumentException("Null key bytes");
+    }
+    this.aesMode = mode;
+    aesKey = new SecretKeySpec(keyBytes, "AES");
+    if (Mode.GCM == mode) {
+      tagLength = GCM_TAG_LENGTH;
+      try {
+        aesCipher = Cipher.getInstance("AES/GCM/NoPadding");
```

Comment:
Should we make "AES/GCM/NoPadding" as a constant string since we use it more than once?

Reply:
Can it be enum?

Reply:
added to Mode enum